src/GHC/Eventlog/Socket.hs

```diff
@@ -6,12 +6,27 @@ module GHC.Eventlog.Socket
     ( startWait, start
     ) where
 
+import Foreign.C
+import Foreign.Ptr
+
 -- | Start listening for eventlog connections, blocking until a client connects.
-startWait :: IO ()
-startWait = c_start True
+startWait :: Maybe FilePath
+          -- ^ File path to the unix domain socket to create. If Nothing, then
+          -- the GHC_EVENTLOG_SOCKET environment variable is used.
+          -> IO ()
+startWait = c_start' True
 
 -- | Start listening for eventlog connections.
-start :: IO ()
-start = c_start False
+start :: Maybe FilePath
+      -- ^ File path to the unix domain socket to create. If Nothing, then
+      -- the GHC_EVENTLOG_SOCKET environment variable is used.
```

Comment:
Can you apply proper Haddock markup here?

Reply:
Looks correct to me and `cabal new-haddock` succeeds. Am I missing something?

Reply:
... I've added "@"s if that's what you meant

Reply:
Technically single-quotes should be used around identifiers so that they are linked. That being said, good enough.